lib/src/elf.ts

```diff
@@ -0,0 +1,99 @@
+// Copyright 2024 Google LLC. Use of this source code is governed by an
+// MIT-style license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+import * as fs from 'fs';
+
+/**
+ * Read chunk of data from a file descriptor into a new Buffer.
+ */
+const readFileDescriptor = function (
+  fd: number,
+  position: number,
+  length: number
+): Buffer {
+  const buffer = Buffer.alloc(length);
+  let offset = 0;
+  while (offset < length) {
+    const bytesRead = fs.readSync(fd, buffer, {
+      offset: offset,
+      position: position + offset,
+    });
+    if (bytesRead === 0) {
+      throw new Error(`failed to read fd ${fd}`);
+    }
+
+    offset += bytesRead;
+  }
+  return buffer;
+};
+
+/**
+ * Parse an ELF file and return its interpreter.
+ */
+export const getELFInterpreter = function (path: string): string {
```

Comment:
Nit:

```suggestion
export function getElfInterpreter(path: string): string {
```

Even in JS code, we try to follow the Dart style of capitalizing 3+-letter acronyms as normal words.